twos-complement/bigint2twos.go

```diff
@@ -0,0 +1,82 @@
+package twoscomplement
+
+import "math/big"
+
+var bigOne = big.NewInt(1)
+
+// ToBytes returns a variable length two's complement byte array representation of the input.
+// Does not alter input.
+func ToBytes(bi *big.Int) []byte {
+	var resultBytes []byte
+	switch bi.Sign() {
+	case -1:
+		// compute 2's complement
+		plus1 := big.NewInt(0)
+		plus1 = plus1.Add(bi, big.NewInt(1)) // add 1
+		resultBytes = plus1.Bytes()
+		for i, b := range resultBytes {
```

Comment:
According to Wikipedia, you should first invert the bytes, then add 1. The inversion and addition steps should not be reversed.

Reply:
Not possible here, because big int doesn't know how to add 1 to a 2's complement, once we invert the bits we're on our own, and I didn't want to also manually implement addition.

It's heavily covered by tests anyway. We can think o a better way at some point.